UPGRADING.md

```diff
@@ -0,0 +1,92 @@
+
+We strongly recommend upgrading the SDK as soon as there is a new version available. Doing so will help you to provide the best Argyle Link experience in your application.
+
+---
+### Upgrade to a new SDK version
+
+`The minimum supported iOS version is iOS 12.`
+
+**Previously installed the Argyle iOS SDK using pod files**
+
+
+1. Run `pod outdated` to see if there is a newer version of the Argyle iOS SDK available. You can also refer to the [release notes](https://github.com/argyle-systems/argyle-link-ios/releases).
+
+2. If a newer version of the SDK is found, update your `Podfile` with this new version. See [Specifying pod versions](https://guides.cocoapods.org/using/the-podfile.html#specifying-pod-versions) to learn how to set a specific version in your `Podfile`. 
+
+3. Run `pod update` to update the SDK with the new version.
+
+4. Make any necessary changes as a result of the upgrade.
+
+5. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+
+---
+
+**Previously installed the Argyle iOS SDK using Carthage**
+
+1. Run `carthage outdated` to see if there is a newer version of the Argyle iOS SDK available. You can also refer to the [release notes](https://github.com/argyle-systems/argyle-link-ios/releases).
+
+2. If a newer version of the SDK is found, update your `Cartfile` with the new version. See [Version requirements](https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#version-requirement) to learn how to set a specific version in your `Cartfile`.
+
+3. Run `carthage update` to update the SDK with the new version.
+
+4. Make any necessary changes as a result of the upgrade.
+
+5. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+---
+
+**Previously installed the Argyle iOS SDK by manually downloading the SDK**
+
+1. Download the current newest SDK and replace the old one.
+
+2. Make any necessary changes as a result of the upgrade.
+
+3. Clean and rebuild your project by selecting **Product > Clean** and then **Product > Build**.
+
+---
+
+
+**To avoid issues and crashes, update your SDK to the latest version before you compile your app. If you manually downloaded your SDK framework locally, you should make sure you are using the latest SDK version.**
+
+---
+
+To verify your SDK version, you can read your `Podfile.lock` and check the `PODS: 'Insert'` value.
+
+---
+# Migrating to Link 4
+
+If you're using a previous major version of Link and would like to migrate to the new version, please update the changed configuration parameters and then follow the usual [iOS Link SDK version upgrade instruction](#upgrade-to-a-new-sdk-version) defined above.
+## Migrating from Link 3
```

Comment:
A bit confusing when `Migrating to Link 4` is in `h1` level, but `3` is in `h2`. Is it because this structure identifies steps as `Migrating to Link 4 from Link 3`?

Reply:
That's how I thought of it. But if this raises questions, maybe it will be simpler without this - only keeping the top heading without mentioning of Link 3.

Reply:
IMHO these two headings could be merged and put under one